File: Pfe_Backend/src/main/java/com/example/demo/entities/Lesson.java
package com.example.demo.entities;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "Lessons")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Lesson {
	@Id
	private String lessonId;
	private Integer duration;
	private String title;
	private String mediaUrl;

}

File: mobile_front/lib/models/lesson.dart
class Lesson {
  final String lessonId;
  final String title;
  final int duration;
  final String? mediaUrl;
  final String? description;

  Lesson({
    required this.lessonId,
    required this.title,
    required this.duration,
    this.mediaUrl,
    this.description,
  });

  factory Lesson.fromJson(Map<String, dynamic> json) {
    return Lesson(
      lessonId: json['lessonId'] ?? '',
      title: json['title'] ?? '',
      duration: json['duration'] ?? 0,
      mediaUrl: json['mediaUrl'],
      description: json['description'],
    );
  }
}
